core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -369,9 +369,9 @@ class ReplicaManagerTest {
       // Use the second instance of metrics group that is constructed. The first instance is constructed by
       // ReplicaManager constructor > BrokerTopicStats > BrokerTopicMetrics.
       val mockMetricsGroup = mockMetricsGroupCtor.constructed.get(1)
-      verify(mockMetricsGroup, times(9)).newGauge(anyString(), any())
-      verify(mockMetricsGroup, times(3)).newMeter(anyString(), anyString(), any(classOf[TimeUnit]))
-      verify(mockMetricsGroup, times(12)).removeMetric(anyString())
+      verify(mockMetricsGroup, times(ReplicaManager.GaugeMetricNames.size)).newGauge(anyString(), any())
```

Comment:
echo'ing a comment that David provided in https://github.com/apache/kafka/pull/13623#discussion_r1182595446, we can replace the `anyString()` with actual names to verify that the argument passed are correct.